File: typescript/src/utils/common/logger.ts
/* Simple logger used inside the parser. */
export const logger = {
    debug: (...args: unknown[]): void => {
        if (process.env.TF_PARSER_DEBUG) {
            console.debug('[parser:debug]', ...args);
        }
    },
    info: (...args: unknown[]): void => {
        console.info('[parser:info]', ...args);
    },
    warn: (...args: unknown[]): void => {
        console.warn('[parser:warn]', ...args);
    }
};

File: python/src/parse_hcl/logger.py
import os
import sys


def _enabled(level: str) -> bool:
    env = os.environ.get("TF_PARSER_DEBUG")
    if not env:
        return level != "debug"
    return True


def debug(*args: object) -> None:
    if _enabled("debug"):
        print("[parser:debug]", *args, file=sys.stderr)


def info(*args: object) -> None:
    if _enabled("info"):
        print("[parser:info]", *args)


def warn(*args: object) -> None:
    if _enabled("warn"):
        print("[parser:warn]", *args, file=sys.stderr)
